sourcegraph/cody
# Fix encoding of file paths in chat links

This fixes https://github.com/sourcegraph/cody/issues/2388, however the encoding format feels odd to me. I've opened https://github.com/microsoft/vscode/issues/200965 to get some clarification about whether this is correct (and if so, hopefully it can be better documented).

https://github.com/sourcegraph/cody/assets/1078012/53893a0b-a5fd-4398-85c7-de95911767b2

@kalanchan 

## Test plan

- Open a chat
- Mention some files with `@`
- Ensure clicking the filenames in the chat message correctly open the files (note: the filenames in the collapsed set are handled differently, the issue was specifically with the converted links in the chat message)
- Also: verify the changes to tests for the new expected format are sensible and match what's discussed in https://github.com/microsoft/vscode/issues/200965 (at the time of writing there has been no response, so we may want to wait for one before merging this?)
